skimage/io/util.py

```diff
@@ -19,10 +19,11 @@ def is_url(filename):
 def file_or_url_context(resource_name):
     """Yield name of file from the given resource (i.e. file or url)."""
     if is_url(resource_name):
-        _, ext = os.path.splitext(resource_name)
+        url_components = urllib.parse.urlparse(resource_name)
+        _, ext = os.path.splitext(url_components.path)
         try:
             with tempfile.NamedTemporaryFile(delete=False, suffix=ext) as f:
```

Comment:
I think this does create a random named temprory file.

It was just taking the whole `.jpeg?laskdjflkasdjflkjasdflkjsad;fj` as the extension